packages/google-vertex/src/google-vertex-image-model.ts

```diff
@@ -0,0 +1,84 @@
+import { ImageModelV1, JSONValue } from '@ai-sdk/provider';
+import { Resolvable, resolve } from '@ai-sdk/provider-utils';
+
+export type GoogleVertexImageModelId =
+  | 'imagen-3.0-generate-001'
+  | 'imagen-3.0-fast-generate-001';
+
+interface GoogleVertexImageModelOptions {
+  provider: string;
+  baseURL: string;
+  headers?: Resolvable<Record<string, string | undefined>>;
+  fetch?: typeof fetch;
+}
+
+// https://cloud.google.com/vertex-ai/generative-ai/docs/image/generate-images
+export class GoogleVertexImageModel implements ImageModelV1 {
+  readonly specificationVersion = 'v1';
+
+  get provider(): string {
+    return this.options.provider;
+  }
+
+  constructor(
+    readonly modelId: GoogleVertexImageModelId,
+    private options: GoogleVertexImageModelOptions,
+  ) {}
+
+  async doGenerate(options: {
+    prompt: string;
+    n: number;
+    size: `${number}x${number}` | undefined;
+    providerOptions: Record<string, Record<string, JSONValue>>;
+    abortSignal?: AbortSignal;
+    headers?: Record<string, string>;
+  }): Promise<{ images: string[] }> {
+    const [width, height] = (options.size ?? '1024x1024')
+      .split('x')
+      .map(Number);
+
+    const response = await (this.options.fetch ?? fetch)(
```

Comment:
should use postJsonToApi similar to `openai-image-model`

Reply:
Done.